lib/tools/adb-commands.js

```diff
@@ -69,11 +69,30 @@ methods.clear = async function (pkg) {
   return await this.shell(['pm', 'clear', pkg]);
 };
 
-methods.grantAllPermissions = async function (pkg) {
-  let permissions = await this.getReqPermissions(pkg);
-  for (let permission of permissions) {
-    await this.grantPermission(pkg, permission);
-  }
+methods.grantAllPermissions = async function (pkg, apk) {
+    let apiLevel = await this.getApiLevel();
+    let targetSdk = null;
+    try {
+      targetSdk = await this.targetSdkVersionFromManifest(apk);
+    } catch (e) {
+      //avoiding logging error stack, as calling library function would have logged
+      log.warn(`Ran into problem getting target SDK version; ignoring...`);
+    }
+    if (apiLevel >= 23 && targetSdk >= 23) {
```

Comment:
shouldn't this look like 

```
apiLevel >= 23 && targetSdk && targetSdk >= 23
```

in such case? What is the result of null >= 23 comparison?